src/stan-users-guide/time-series.qmd

```diff
@@ -634,334 +634,116 @@ time of the original iterations.
+$$
+ \rho_k = p(z_0 = k \mid \phi).
 $$
-In short, HMMs form a discrete mixture model where the mixture
-component indicators form a latent Markov chain.
-
-
-
-### Supervised parameter estimation {-}
 
-In the situation where the hidden states are known, the following
-naive model can be used to fit the parameters $\theta$ and $\phi$.
+As an example, consider a three-state model, with $K = 1, 2, 3$.
+The observations are normally distributed with
+$$
+  y_n \sim \text{normal}(\mu_k, \sigma),
+$$
+where $\mu = (1, 5, 9)$ and the standard deviation $\sigma$ is the same
+across all observation.
+The model is then
 
 ```stan
 data {
-  int<lower=1> K;             // num categories
-  int<lower=1> V;             // num words
-  int<lower=0> T;             // num instances
-  array[T] int<lower=1, upper=V> w;  // words
-  array[T] int<lower=1, upper=K> z;  // categories
-  vector<lower=0>[K] alpha;   // transit prior
-  vector<lower=0>[V] beta;    // emit prior
-}
-parameters {
-  array[K] simplex[K] theta;        // transit probs
-  array[K] simplex[V] phi;          // emit probs
-}
-model {
-  for (k in 1:K) {
-    theta[k] ~ dirichlet(alpha);
-  }
-  for (k in 1:K) {
-    phi[k] ~ dirichlet(beta);
-  }
-  for (t in 1:T) {
-    w[t] ~ categorical(phi[z[t]]);
-  }
-  for (t in 2:T) {
-    z[t] ~ categorical(theta[z[t - 1]]);
-  }
+  int N;  // Number of observations
+  array[N] real y;
 }
-```
 
-Explicit Dirichlet priors have been provided for $\theta_k$ and
-$\phi_k$; dropping these two statements would implicitly take the
-prior to be uniform over all valid simplexes.
-
-### Start-state and end-state probabilities {-}
-
-Although workable, the above description of HMMs is incomplete because
-the start state $z_1$ is not modeled (the index runs from 2 to $T$).
-If the data are conceived as a subsequence of a long-running process,
-the probability of $z_1$ should be set to the stationary state
-probabilities in the Markov chain.  In this case, there is no distinct
-end to the data, so there is no need to model the probability that the
-sequence ends at $z_T$.
-
-An alternative conception of HMMs is as models of finite-length
-sequences.  For example, human language sentences have distinct
-starting distributions (usually a capital letter) and ending
-distributions (usually some kind of punctuation).  The simplest way to
-model the sequence boundaries is to add a new latent state $K+1$,
-generate the first state from a categorical distribution with
-parameter vector $\theta_{K+1}$, and restrict the transitions so that
-a transition to state $K+1$ is forced to occur at the end of the
-sentence and is prohibited elsewhere.
-
-### Calculating sufficient statistics {-}
-
-The naive HMM estimation model presented above can be sped up
-dramatically by replacing the loops over categorical distributions
-with a single multinomial distribution.
+parameters {
+  // Rows of the transition matrix
+  array[3] simplex[3] gamma_arr;
```

Comment:
Ack.  Does this mean you require an array of simplexes, but our `row_stochastic_matrix` type is a matrix so it can't be used.  If so, that's a real bummer, because nobody will ever use the `row_stochastic_matrix` type!  I guess we didn't think the data types through with @SteveBronder.

Reply:
We wrote the HMM suites before we had stochastic matrices. We can do a PR and change the arguments.